shared/src/business/entities/Statistic.test.js

```diff
@@ -135,4 +179,154 @@ describe('Statistic', () => {
       ]);
     });
   });
+
+  describe('Penalties', () => {
+    let statistic;
+    let statisticId = applicationContext.getUniqueId();
+    let penaltyArrayLength;
+    const MOCK_PENALTY_WITH_STATISTIC_ID = {
+      entityName: 'Penalty',
+      name: 'Penalty 1 (IRS)',
+      penaltyAmount: 200,
+      penaltyId: '081108f8-8b01-4e49-b437-781a581a16ac',
+      penaltyType: 'irsPenaltyAmount',
+      statisticId,
+    };
+    const MOCK_PENALTY_WITHOUT_STATISTIC_ID = {
+      entityName: 'Penalty',
+      name: 'Penalty 1 (Court)',
+      penaltyAmount: 200,
+      penaltyId: '081108f8-8b01-4e49-b437-781a581a16ac',
+      penaltyType: 'determinationPenaltyAmount',
+    };
+    const MOCK_UPDATED_PENALTY = {
+      entityName: 'Penalty',
+      name: 'I am an updated penalty!',
+      penaltyAmount: 250,
+      penaltyId: '123408f8-8b01-4e49-b437-123a581a12bb',
+      penaltyType: 'irsPenaltyAmount',
+      statisticId,
+    };
+
+    beforeEach(() => {
+      statistic = new Statistic(
+        {
+          irsDeficiencyAmount: 1,
+          irsTotalPenalties: 1,
+          penalties: [
+            {
+              irsPenaltyAmount: 100.0,
+              name: 'Penalty 1',
+              penaltyId: '123408f8-8b01-4e49-b437-123a581a12bb',
+              statisticId,
+            },
+          ],
+          statisticId,
+          year: '2001',
+          yearOrPeriod: 'Year',
+        },
+        { applicationContext },
+      );
+      penaltyArrayLength = statistic.penalties.length;
+    });
+
+    it('should add a penalty with a statistics id to the penalties array', () => {
+      statistic.addPenalty({
+        applicationContext,
+        rawPenalty: MOCK_PENALTY_WITH_STATISTIC_ID,
+      });
+      expect(statistic.penalties.length).toEqual(penaltyArrayLength + 1);
+      expect(statistic.penalties[1]).toEqual(MOCK_PENALTY_WITH_STATISTIC_ID);
+    });
+
+    it('should add a penalty without a statistics id to the penalties array', () => {
+      statistic.addPenalty({
+        applicationContext,
+        rawPenalty: MOCK_PENALTY_WITHOUT_STATISTIC_ID,
+      });
+
+      expect(statistic.penalties.length).toEqual(penaltyArrayLength + 1);
+      expect(statistic.penalties[1]).toEqual({
+        ...MOCK_PENALTY_WITHOUT_STATISTIC_ID,
+        statisticId,
+      });
+    });
+
+    it('should update the penalty in the penalties array', () => {
+      expect(statistic.penalties.length).toEqual(penaltyArrayLength);
+
+      statistic.updatePenalty(MOCK_UPDATED_PENALTY);
+
+      expect(statistic.penalties.length).toEqual(penaltyArrayLength);
+      expect(statistic.penalties[0]).toEqual(MOCK_UPDATED_PENALTY);
+    });
+
+    it('should itemize penalties created prior to penalty itemization', () => {
```

Comment:
This test and the one following it should have different names, based on which type of existing penalty we are itemizing

Reply:
Done